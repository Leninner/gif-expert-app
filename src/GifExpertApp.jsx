import CssBaseline from '@mui/material/CssBaseline';
import { useState } from 'react';
import { List } from '@mui/material';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultState = [] }) => {
  const [categories, setCategories] = useState(defaultState);

  const handleAddCategory = (value) => {
    setCategories([value, ...categories]);
  };

  return (
    <>
      <CssBaseline />
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={handleAddCategory} />
      <List>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </List>
    </>
  );
};
