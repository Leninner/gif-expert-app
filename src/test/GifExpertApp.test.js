import { GifExpertApp } from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('GifExpertApp', () => {
  test('should match with the snapshot', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show a list', () => {
    const categories = ['One Punch', 'Samurai X'];
    const wrapper = shallow(<GifExpertApp defaultState={categories} />);

    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
