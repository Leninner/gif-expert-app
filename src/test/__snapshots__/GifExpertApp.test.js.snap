// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should match with the snapshot 1`] = `
<Fragment>
  <CssBaseline />
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ForwardRef(List) />
</Fragment>
`;
